inc/dashboard/namespace.php

```diff
@@ -0,0 +1,436 @@
+ *
+ * Note: This adds an additional span around the column title as there is no current way to filter the column title span itself.
+ *
+ * @see https://github.com/WordPress/WordPress/blob/643888dfc6697877609672087b4d3b4e5bf545ff/wp-admin/includes/class-wp-posts-list-table.php#L1059
+ *
+ * @param array $xb_columns An array of columns for the XB post type.
+ *
+ * @return array The filtered array of columns.
+ */
+function add_microcopy_to_column_titles( array $xb_columns ) : array {
+	$microcopy = [
+		'block' => __( 'List of XBs modified in the selected date range', 'altis-analytics' ),
+		'views' => __( 'Total unique views of the XBs during the selected date range', 'altis-analytics' ),
+		'conversion' => __( 'Average conversion is calculated as the total conversion expressed as a % of total unique views of the XB during the selected date range', 'altis-analytics' ),
+		'details' => __( 'Block last modified date and author', 'altis-analytics' ),
+	];
+
+	foreach ( $xb_columns as $column => $title ) {
+		$xb_columns[ $column ] = '<span title="' . esc_html( $microcopy[ $column ] ) . '">' . esc_html( $title ) . '</span>
+			<span class="screen-reader-text">' . esc_html( $microcopy[ $column ] ) . '</span>';
+	}
+
+	return $xb_columns;
+}
+
+/**
+ * Update the Block column's orderby value.
+ *
+ * This displays the title, but the column is a custom column called "block", so we need to change the orderby value to "title".
+ *
+ * @param array $vars An array of query variables passed in from the filter.
+ *
+ * @return array The updated query variables.
+ */
+function xb_block_column_orderby( $vars ) : array {
+	if ( isset( $vars['orderby'] ) && 'block' === $vars['orderby'] ) {
+		$vars['orderby'] = 'title';
+	}
+
+	return $vars;
+}
+
+/**
+ * Render links for date range data.
+ */
+function render_date_range_links() {
+	$ranges = [
+		[
+			'days' => 7,
+			'label' => __( '7 days', 'altis-analytics' ),
+		],
+		[
+			'days' => 30,
+			'label' => __( '30 days', 'altis-analytics' ),
+		],
+		[
+			'days' => 90,
+			'label' => __( '90 days', 'altis-analytics' ),
+		],
+	];
+
+	$days = get_days_view();
+	?>
+	<div class="xb-analytics-date-range">
+		<?php
+		foreach ( $ranges as $current ) {
+			// Compare the current days value with the current days value pulled from the url query variables.
+			if ( $days === $current['days'] ) {
+				$selected = 'selected';
+			} else {
+				$selected = '';
+			}
+			?>
+			<a href="<?php echo esc_url_raw( add_query_arg( [ 'days' => $current['days'] ] ) ); ?>" class="<?php echo esc_html( "date-range-button $selected" ); ?>"><?php echo esc_html( $current['label'] ); ?></a>
+		<?php } ?>
+	</div>
+	<?php
+}
+
+/**
+ * Render the block name column data.
+ */
+function render_block_column() {
+	global $post;
+	?>
+	<strong><a href="<?php echo esc_url_raw( admin_url( '/admin.php?page=xb-analytics&post=' . $post->ID ) ); ?>"><?php echo esc_attr( $post->post_title ); ?></a></strong>
```

Comment:
I'd like to chat through how the XB post types are hierarchically linked, I'm not sure I'm clear on that yet.

Reply:
Pretty much just this https://github.com/humanmade/aws-analytics/blob/master/inc/blocks/namespace.php#L112

The parent is the post the XB was created on.

Reply:
After looking at this filter, I think it's better to pass on it here. The `$context` parameter doesn't really help us out as much as I would like it to and the two places we could use it (here, to link to the analytics view and in the quick edit row, to link to the parent post) are two very different destinations. If `$context` gave us actually useful information, like where the link was being used or something, this would be brilliant, but as it is, for both those links it just returns `'display'` which is totally unhelpful.

That said, we probably _should_ still add the filter -- and always link back to the parent post -- and just not use it on this line of code, specifically, so that we never end up on a edit single block page.